config/accounts_test.go

```diff
@@ -8,64 +8,64 @@ import (
 )
 
-func TestAccountGDPREnabledForIntegrationType(t *testing.T) {
+func TestAccountGDPREnabledForChannelType(t *testing.T) {
 	trueValue, falseValue := true, false
 
 	tests := []struct {
-		description         string
-		giveIntegrationType IntegrationType
-		giveGDPREnabled     *bool
-		giveWebGDPREnabled  *bool
-		wantEnabled         *bool
+		description        string
+		giveChannelType    ChannelType
+		giveGDPREnabled    *bool
+		giveWebGDPREnabled *bool
+		wantEnabled        *bool
 	}{
 		{
-			description:         "GDPR Web integration enabled, general GDPR disabled",
-			giveIntegrationType: IntegrationTypeWeb,
-			giveGDPREnabled:     &falseValue,
-			giveWebGDPREnabled:  &trueValue,
-			wantEnabled:         &trueValue,
+			description:        "GDPR Web channel enabled, general GDPR disabled",
+			giveChannelType:    ChannelWeb,
+			giveGDPREnabled:    &falseValue,
+			giveWebGDPREnabled: &trueValue,
+			wantEnabled:        &trueValue,
 		},
 		{
-			description:         "GDPR Web integration disabled, general GDPR enabled",
-			giveIntegrationType: IntegrationTypeWeb,
-			giveGDPREnabled:     &trueValue,
-			giveWebGDPREnabled:  &falseValue,
-			wantEnabled:         &falseValue,
+			description:        "GDPR Web channel disabled, general GDPR enabled",
+			giveChannelType:    ChannelWeb,
+			giveGDPREnabled:    &trueValue,
+			giveWebGDPREnabled: &falseValue,
+			wantEnabled:        &falseValue,
 		},
 		{
-			description:         "GDPR Web integration unspecified, general GDPR disabled",
-			giveIntegrationType: IntegrationTypeWeb,
-			giveGDPREnabled:     &falseValue,
-			giveWebGDPREnabled:  nil,
-			wantEnabled:         &falseValue,
+			description:        "GDPR Web channel unspecified, general GDPR disabled",
+			giveChannelType:    ChannelWeb,
+			giveGDPREnabled:    &falseValue,
+			giveWebGDPREnabled: nil,
+			wantEnabled:        &falseValue,
 		},
 		{
-			description:         "GDPR Web integration unspecified, general GDPR enabled",
-			giveIntegrationType: IntegrationTypeWeb,
-			giveGDPREnabled:     &trueValue,
-			giveWebGDPREnabled:  nil,
-			wantEnabled:         &trueValue,
+			description:        "GDPR Web channel unspecified, general GDPR enabled",
+			giveChannelType:    ChannelWeb,
+			giveGDPREnabled:    &trueValue,
+			giveWebGDPREnabled: nil,
+			wantEnabled:        &trueValue,
 		},
 		{
-			description:         "GDPR Web integration unspecified, general GDPR unspecified",
-			giveIntegrationType: IntegrationTypeWeb,
-			giveGDPREnabled:     nil,
-			giveWebGDPREnabled:  nil,
-			wantEnabled:         nil,
+			description:        "GDPR Web channel unspecified, general GDPR unspecified",
+			giveChannelType:    ChannelWeb,
+			giveGDPREnabled:    nil,
+			giveWebGDPREnabled: nil,
+			wantEnabled:        nil,
 		},
 	}
 
 	for _, tt := range tests {
 		account := Account{
 			GDPR: AccountGDPR{
 				Enabled: tt.giveGDPREnabled,
-				IntegrationEnabled: AccountIntegration{
+				IntegrationEnabled: AccountChannel{
```

Comment:
This test case is still currently setting the `IntegrationEnabled` field when it should be setting the `ChannelEnabled` field. We also probably need a couple of additional test cases for when the deprecated `IntegrationEnabled` field is set but `ChannelEnabled` is not and also to verify precedence when both fields are set.
If you run code coverage you should see that we don't have full coverage of the function.